models/mongo/members.go

```diff
@@ -0,0 +1,143 @@
+package mongo
+
+import (
+	"accounts-service/auth"
+	"accounts-service/models"
+	"errors"
+
+	"github.com/google/uuid"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.uber.org/zap"
+	"golang.org/x/net/context"
+)
+
+type membersRepository struct {
+	logger *zap.Logger
+	db     *mongo.Database
+	coll   *mongo.Collection
+}
+
+func NewMembersRepository(db *mongo.Database, logger *zap.Logger) models.MembersRepository {
+	rep := &membersRepository{
+		logger: logger.Named("mongo").Named("members"),
+		db:     db,
+		coll:   db.Collection("members"),
+	}
+
+	_, err := rep.coll.Indexes().CreateOne(
+		context.Background(),
+		mongo.IndexModel{
+			Keys:    bson.D{{Key: "account_id", Value: 1}, {Key: "group_id", Value: 1}},
+			Options: options.Index().SetUnique(true),
+		},
+	)
```

Comment:
Just curious, is this operation idempotent?

Reply:
i think so, mongo does not allowed dupplicate index, so it's only created once